docs/development-setup.md

```diff
@@ -1,73 +1,137 @@
+$ docker build --no-cache --tag=buildroot .
 ```
 
-Then you can start OpenShift with one simple command:
+If you are using `osbs-client` from git master branch, you should also install
+git master of atomic-reactor:
 
 ```
-$ docker-compose up -d
+$ docker build --no-cache --tag=buildroot --build-arg LATEST_CODE=true .
 ```
 
 
-## setup without docker-compose
+## Building images, finally
+
+In order to submit a build, you need to have a permission. If you started
+OpenShift with `oc cluster up`, there's a user `developer` set up with
+namespace `myproject`, this is what we'll be using.
 
-You don't need docker-compose to launch the container (it's just convenience); you can run the OpenShift container like this:
+
+### Obtaining token
+
+OpenShift uses Oauth tokens for authentication, You can easily get token of
+currently logged-in user:
 
 ```
-$ docker run -d --name "origin" \
-  --privileged --pid=host --net=host \
-  -v /:/rootfs:ro -v /var/run:/var/run:rw -v /sys:/sys -v /var/lib/docker:/var/lib/docker:rw \
-  -v /var/lib/openshift/openshift.local.volumes:/var/lib/openshift/openshift.local.volumes \
-  openshift/origin:v1.0.6 start
+$ oc whoami -t
+hb1WN2Tx8yV4s4slFxhSRm24Hk_Pwma5wZiW0iadP4c
 ```
 
+We then feed this token to config.
 
-## building images
 
-In order to successfully build an image you need to have 3 things:
+### Config file
 
-1. permission to submit a build — for development it's fine to do this (never do that on a public system!):
-    ```
-    $ docker exec -ti $openshift_container bash
-    $ oadm policy add-role-to-group edit system:unauthenticated system:authenticated
-    ```
+`osbs-client` accepts configuration from CLI and from an ini-style
+configuration file. Here's a really simple one you can use as a start:
 
+```
+[local]
+openshift_uri = https://localhost:8443/
+registry_uri = <registry:5000>
+namespace = myproject
+use_kerberos = false
+verify_ssl = false
+use_auth = true
+token = hb1WN2Tx8yV4s4slFxhSRm24Hk_Pwma5wZiW0iadP4c
+```
 
-2. buildroot image — see [this section](https://github.com/projectatomic/osbs-client/blob/master/docs/osbs_instance_setup.md#atomic-reactor) section in [deplyoment guide](https://github.com/projectatomic/osbs-client/blob/master/docs/osbs_instance_setup.md)
+Copy the content and place it to a file named `osbs.conf`.
 
-3. make OpenShift API accessible so build can submit metadata once build is done:
-    ```
-    $ firewall-cmd --permanent --add-port 8443/tcp
-    $ firewall-cmd --add-port 8443/tcp
-    ```
 
-4. config for osbs-client; something like this should do
-    ```
-    [local]
-    openshift_uri = https://172.17.42.1:8443/
-    sources_command = fedpkg sources
-    registry_uri = <registry:5000>
-    vendor = <this_company>
-    authoritative_registry = <registry.example.com>
-    build_host = <builder.example.com>
-    architecture = x86_64
-    verify_ssl = false
-    builder_use_auth = false
-    use_auth=false
-    ```
+#### Registry
```

Comment:
Shouldn't this be moved after the getting the token step? The registry IP should be used in `osbs.conf`.